### PR TITLE
Add a linter to prevent use of os.Getenv and os.LookupEnv

### PR DESCRIPTION
**Pull request by geeknoid**
- Add more unit tests to pkg/env to bring coverage to 100%

- Move existing linter sources from test/util/checker to tools/checker

- Now log when the same env var is registered multiple times using different metadata.

- Add a comment about some commented-out chunk of code, as a follow-on to a previous PR.

### CODE REVIEW

File: tools/checker/testlinter/testdata/unit_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func TestInvalidShort(t *testing.T) {
 
 	if testing.Short() {
 		t.Skip("skipping uint test in short mode.")
-	}
+	}F
```
Comment (nmittler) [inline tools/checker/testlinter/testdata/unit_test.go]: type-o?

File: pkg/env/var_test.go [hunk 2]
```diff
@@ -43,6 +43,11 @@ func TestString(t *testing.T) {
 		t.Errorf("Expected not present")
 	}
 
+	v = ev.Get()
+	if v != "123" {
+		t.Errorf("Expected 123, got %s", v)
```
Comment (nmittler) [inline pkg/env/var_test.go]: do these error message pass the linter? I thought they had to start with lowercase?

Comment (ozevren) [reply pkg/env/var_test.go]: linter doesn't care about the testing.T calls.

File: tools/checker/envvarlinter/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# envvarlinter
+
+envvarlinter ensures that non-test files don't use os.Getenv and os.LookupEnv and instead use the functions from pkg/env.
+
+envvarlinter is baed on [Checker](../README.md), and this package provides the [custom rules](rules) implementation.
```
Comment (howardjohn) [inline tools/checker/envvarlinter/README.md]: typo "baed"

File: tools/checker/envvarlinter/rules/no_os_env.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+// GetID returns skip_by_issue_rule.
+func (lr *NoOsEnv) GetID() string {
+	return GetCallerFileName()
+}
+
+// Check verifies there are no calls to os.Getenv or os.LookupEnv
+func (lr *NoOsEnv) Check(aNode ast.Node, fs *token.FileSet, lrp *checker.Report) {
+	if ce, ok := aNode.(*ast.CallExpr); ok {
+		if MatchCallExpr(ce, "os", "Getenv") {
```
Comment (kyessenov) [inline tools/checker/envvarlinter/rules/no_os_env.go]: nit: this does not catch renamed packages (`import x os`), but let's keep that a secret :)

**Pull request merged**